src/elfpy/pricing_models/base.py

```diff
@@ -375,79 +378,24 @@ def get_max_long(
                 market_state=market_state,
                 time_remaining=time_remaining,
-            )
-            maybe_max_long = trade_result.breakdown.with_fee
-
-            # If the max long is less than or equal to zero, we need to reduce
-            # the amount of bonds that we are attempting to purchase. Otherwise,
-            # go through the other checks for the bisection.
-            if maybe_max_long <= 0:
-                bond_percent -= step_size
-            else:
-                # TODO: Do we actually need to do this? Run some tests and see
-                # if the result that only uses calc_in_given_out is always
-                # slightly lower. If so, then just take that.
-                #
-                # Compute the amount of bonds that will be purchased with the
-                # amount of base. We feed the result of calc_in_given_out back
-                # into the pricing model as this computation is more accurate.
-                trade_result = self.calc_out_given_in(
-                    in_=Quantity(amount=maybe_max_long, unit=TokenType.BASE),
-                    market_state=market_state,
-                    time_remaining=time_remaining,
-                )
-                d_bonds = trade_result.breakdown.with_fee
-
-                # Apply the trade to the market state.
-                market_state_post_trade = copy.copy(market_state)
-                market_state_post_trade.apply_delta(
-                    delta=MarketDeltas(
-                        d_base_asset=trade_result.market_result.d_base,
-                        d_token_asset=trade_result.market_result.d_bonds,
-                        d_base_buffer=d_bonds,
-                    )
-                )
-
-                # Verify that none of the reserve invariants were broken. If
-                # the invariants were broken, reduce the bond size. Otherwise,
-                # we've found a new max long amount, so we store that value and
-                # proceed with bisection with larger bond purchases.
-                if (
-                    market_state.bond_reserves < d_bonds
-                    or self.calc_apr_from_reserves(market_state=market_state_post_trade, time_remaining=time_remaining)
-                    < 0
-                    or market_state_post_trade.share_price * market_state_post_trade.share_reserves
-                    < market_state_post_trade.base_buffer
-                    or market_state_post_trade.bond_reserves < market_state_post_trade.bond_buffer
-                ):
-                    bond_percent -= step_size
-                else:
-                    last_maybe_max_long = (maybe_max_long, d_bonds)
-                    if bond_percent == 1:
-                        return last_maybe_max_long
-                    bond_percent += step_size
-
-        return last_maybe_max_long
+            ).breakdown.with_fee
+            - market_state.base_buffer / market_state.share_price
```

Comment:
Is this -c * z a hyperdrive thing? this change is in `base` -- would that break for yieldspace?